create-ocp-project/create-ocp-project.sw.yaml

```diff
@@ -52,35 +56,37 @@ states:
           refName: jiraCreateIssue
           arguments:
             fields:
-              description: '"Please review request to create project " + .inputs.projectName'
+              description: '"Please review request to create project " + .projectName'
               issuetype:
                 name: Task
               labels:
                 - backstage-workflow
                 - '"workflowId=" + $WORKFLOW.instanceId'
               project:
-                key: .inputs.operationsProjectKey
-              summary: '"Request authorization to create project " + .inputs.projectName'
+                key: .operationsProjectKey
+              summary: '"Request authorization to create project " + .projectName'
         actionDataFilter:
           toStateData: .operationsJiraCreateIssueResult
     transition: "Notify Backstage: Jira ticket created"
   - name: "Notify Backstage: Jira ticket created"
     type: operation
     actions:
       - name: "Create notification: Jira ticket created"
+        condition: .backstageUser != null or .backstageGroup != null
         functionRef:
           refName: createNotification
           arguments:
-            title: "Jira ticket created"
-            message: "Jira ticket created"
-            origin: "SonataFlow"
-            topic: "Create OCP Project workflow"
-            targetUsers:
-              - "default/guest" # replace with the approver user
-            actions:
-              - title: "View on Jira"
-                # There are more dynamic ways to get the Jira URL, but for simplicity, we are using a hardcoded URL
-                url: '$SECRET.jira_url + "/jira/servicedesk/projects/" + .inputs.operationsProjectKey + "/issues/" + .operationsJiraCreateIssueResult.key'
+            recipients:
+              type: "entity"
+              entityRef:
+                - "${ .backstageUser != null ? 'user:default/' + .backstageUser : '' }"
```

Comment:
I think we should not use backstageUser/backstageGroup. instead, use entityRef in input. we rely here on the fact the notifications plugin will accept an empty entityRef in the array. This is wrong.
anyhow...here's a code that does the same and compiles:
'if .backstageUser != null then "user:default/" + .backstageUser else "" end'
replace everything after '- ' with that
@pkliczewski

Reply:
however ,this will still leave one empty entry (when tried that with curl, there wasn't an issue, but perhaps it makes more sense to use entityRef as input parameter here. let me try and adapt the code.